### ISSUE TITLE
Unable to build pdf of documentation

### ISSUE THREAD

**Issue by j605** [JuliaLang/julia#29694]
From https://discourse.julialang.org/t/offline-julia-1-0-documentation/13263/8?u=j605

I tried to build the PDF as instructed in the post but it fails. I am using the current release tarball but @tpapp mentions it fails in master as well.

Navigate to docs directory and running `make.jl`
```julia
julia make.jl pdf
   Cloning default registries into /home/jagan/srces/julia/repos/community-x86_64/src/julia/doc/deps/registries
   Cloning registry General from "https://github.com/JuliaRegistries/General.git"
  Updating registry at `~/srces/julia/repos/community-x86_64/src/julia/doc/deps/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
 Installed DocStringExtensions ─ v0.4.5
 Installed Compat ────────────── v1.0.0
 Installed Documenter ────────── v0.19.1
Documenter: setting up build directory.
Documenter: expanding markdown templates.
 !! Warning in src/manual/unicode-input.md:

Failed to evaluate `@eval` block.

ERROR: SystemError("opening file /doc/UnicodeData.txt", 2, nothing)

Documenter: building cross-references.
Documenter: running document checks.
 !! Skipped doctesting.
 > checking footnote links.
Documenter: populating indices.
ERROR: LoadError: `makedocs` encountered an error. Terminating build
Stacktrace:
 [1] runner(::Type{Documenter.Builder.RenderDocument}, ::Documenter.Documents.Document) at /home/jagan/srces/julia/repos/community-x86_64/src/julia/doc/deps/packages/Documenter/Psnle/src/Builder.jl:203
 [2] dispatch(::Type{Documenter.Builder.DocumentPipeline}, ::Documenter.Documents.Document) at /home/jagan/srces/julia/repos/community-x86_64/src/julia/doc/deps/packages/Documenter/Psnle/src/Selectors.jl:168
 [3] #2 at /home/jagan/srces/julia/repos/community-x86_64/src/julia/doc/deps/packages/Documenter/Psnle/src/Documenter.jl:204 [inlined]
 [4] cd(::getfield(Documenter, Symbol("##2#3")){Documenter.Documents.Document}, ::String) at ./file.jl:96
 [5] #makedocs#1 at /home/jagan/srces/julia/repos/community-x86_64/src/julia/doc/deps/packages/Documenter/Psnle/src/Documenter.jl:203 [inlined]
 [6] (::getfield(Documenter, Symbol("#kw##makedocs")))(::NamedTuple{(:build, :modules, :clean, :doctest, :linkcheck, :linkcheck_ignore, :strict, :checkdocs, :format, :sitename, :authors, :analytics, :pages, :html_prettyurls, :html_canonical, :assets),Tuple{String,Array{Module,1},Bool,Bool,Bool,Array{String,1},Bool,Symbol,Symbol,String,String,String,Array{Pair{String,Any},1},Bool,Nothing,Array{String,1}}}, ::typeof(makedocs)) at ./none:0
 [7] top-level scope at none:0
in expression starting at /home/jagan/srces/julia/repos/community-x86_64/src/julia/doc/make.jl:150
```

**Comment by cossio** [comment]
I think this issue should be tracked at https://github.com/JuliaDocs/Documenter.jl.

**Comment by mortenpi** [comment]
The issue appears to be related to how Documenter is set up here, so I'd say it's appropriate here.

The problem seems to be related to the `unicode_data()` function in an at-eval block on the [manual/unicode-input.md](https://github.com/JuliaLang/julia/blob/master/doc/src/manual/unicode-input.md) page. Do you run into a similar error when you build the HTML locally?

**Comment by j605** [comment]
This might be because I am using system installed (from Arch Linux) Julia but this is what I get,
```julia
julia> Sys.BINDIR
"/usr/bin"

julia> normpath(Sys.BINDIR, "..", "..", "doc", "UnicodeData.txt")
"/doc/UnicodeData.txt"
```

**Comment by mortenpi** [comment]
I have an idea for a fix (22f697458effbdc3148b06c9c657bef9ecbad7ea). But actually the docs should be built with `make` anyway. And if you haven't built the `julia` binary in the source tree, it will fail with an error saying that it is missing.

And that is good, since if you use an "external" Julia and call `make.jl` directly, then the docstrings may be out of sync with the manual pages (i.e. when the "external" Julia version is different from the source tree). So calling `make.jl` directly should actually be discouraged.

**Comment by j605** [comment]
I was using the same Julia as the release tarball though. Building Julia to get the documentation might be too difficult. As mentioned in the OP, a PDF of the manual on the downloads page would be great. 

Is there a reason why building the documentation is coupled to building Julia itself? I will close the issue if this out of scope, thanks.

**Comment by fredrikekre** [comment]
Probably fixed by #30012, `make -C doc pdf platform=docker` can be used to build the tex using a docker image (https://cloud.docker.com/u/juliadocs/repository/docker/juliadocs/documenter-latex) that has all required tex installs and fonts available.

**Comment by j605** [comment]
I was able to build the documentation by using https://github.com/JuliaLang/julia/issues/29694#issuecomment-448188200 without docker. I already had the whole texlive distribution installed in Arch Linux and I symlinked the julia binary from `/usr/bin/julia` which just worked. Thanks, now I can build the pdf without any hassle :)
